### PR TITLE
aqua blog post script 

### PR DESCRIPTION
**Pull request by happycontribute**
*Description of changes:*
I am trying to put scripts for AQUA blog post in DB Eng repository and refer it in my blogpost. Public and Customer can follow blogpost and use scripts from this public repository 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/BlogContent/AQUA-Evaluation/aqua_capture_query.sh [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+#           -p PGPORT
+#           -U PGUSR
+#           -d PGDATABASE
+#           -s STARTTIME (FORMAT: YYYY-MM-DD hh:mm:ss)
+#           -e ENDTIME (FORMAT: YYYY-MM-DD hh:mm:ss)
+#           -t PGCONNECT_TIMEOUT
+# sample run with parameters below :
+#./aqua_capture_query.sh -h 34.226.44.240 -p 5439 -d dev -U test_user -s "2021-09-06 00:00:00" -e "2021-09-09 23:00:00"
+#######################################################################################################################
+
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_capture_query.sh:17]: Remove extra space

File: src/BlogContent/AQUA-Evaluation/aqua_capture_query.sh [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+date +"%Y-%m-%d %T" > date.txt
+echo 'date' >> dates.txt
+
+if [[ -z $PGHOST ]] || [[ -z $PGPORT ]] || [[ -z $PGDATABASE ]] || [[ -z $PGUSER ]] ; then
+      echo "connection parameters required and cannot be empty, please enter correct  hostname, dbname, port-number, username to connect"
+      exit -1
+fi
+
+#"If user doesn't provide start and endtime, default to current time as start time and use queries from first 3 hours."
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_capture_query.sh:49]: Clean up quotes/spacing

File: src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+# Arguments are: 
+#           -h PGHOST
+#           -p PGPORT
+#           -U PGUSR
+#           -d PGDATABASE
+#           -t PGCONNECT_TIMEOUT
+# sample run with parameters below :
+#./aqua_execute_query.sh -h 3.88.94.167 -p 5439 -d dev -U test_user 
+#######################################################################################################################
+
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh:15]: Remove extra space

File: src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+done
+
+read -s -p "Password to connect to Redshift cluster: " PGPASSWORD
+export  PGPASSWORD=$PGPASSWORD
+echo
+if [[ -z $PGCONNECT_TIMEOUT ]]; then
+  export PGCONNECT_TIMEOUT=10
+fi
+
+#TODO: Validate ALL inputs as shown below
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh:37]: Remove todo or implement.

File: src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+if [[ -z $PGHOST ]] || [[ -z $PGPORT ]] || [[ -z $PGDATABASE ]] || [[ -z $PGUSER ]] ; then
+echo "connection parameters required and cannot be empty, please enter correct  hostname, dbname, port-number, username to connect"     
+   exit -1
+fi
+
+if [ ! -s ./capture.sql  ]; then
+
+echo "The capture.sql seems empty, please capture aqua eligible queries using aqua_capture_query.sh then execute aqua_execute_query.sh"
+else
+#log execution startime
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh:48]: Is there a reason there is no indentation for these scripts?  Can you properly indent for readability?

File: src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+   exit -1
+fi
+
+if [ ! -s ./capture.sql  ]; then
+
+echo "The capture.sql seems empty, please capture aqua eligible queries using aqua_capture_query.sh then execute aqua_execute_query.sh"
+else
+#log execution startime 
+date +"%Y-%m-%d %T" > date.txt 
+#execute in redhshift three times
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh:50]: typo

File: src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+echo $iter
+psql -q -f ./capture_tmp.sql -o cap.out > sql_error.log  2>&1 
+CONN_STATUS=$?
+if [ $CONN_STATUS -eq 2 ]; then
+echo "Connection failed. Please try again with correct connection parameters"
+exit $CONN_STATUS
+fi
+SQL_ERROR=$(cat sql_error.log |  grep -i "error\|failed\|timeout" | wc -l )
+if [ $SQL_ERROR -gt 1 ]; then
+echo "Please check the sql_error.log for erros, once SQL error's fixed and then re-run aqua_execute_query.sh"
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh:65]: typo

File: src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+psql -q  -f ./tmp_capture.sql  -o cap.out  > sql_error.log  2>&1 
+CONN_STATUS=$?
+if [ $CONN_STATUS -eq 2 ]; then
+echo "Connection failed. Please try again with correct connection parameters"
+exit $CONN_STATUS
+fi
+#cat sql_error.log |  grep -i -q "error\|failed\|timeout"
+SQL_ERROR=$(cat sql_error.log |  grep -i -q "error\|failed\|timeout" | wc -l )
+if [ $SQL_ERROR -gt 1  ]; then
+echo "Please check the sql_error.log for erros, once SQL error's fixed and then  re-run aqua_execute_query.sh"
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh:86]: typo

File: src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+rm -f sql_error.log
+exit $SQL_ERROR 
+fi
+done
+
+ 
+rm -f  capture_tmp.sql
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh:93]: extra space

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+#           -h PGHOST
+#           -p PGPORT
+#           -U PGUSR
+#           -d PGDATABASE
+#           -s STARTTIME (FORMAT: YYYY-MM-DD HH:MI:SS)
+#           -e ENDTIME (FORMAT: YYYY-MM-DD HH:MI:SS)
+## sample run with parameters below :
+# ./aqua_perf_compare.sh -h 3.88.94.167 -p 5439 -d dev -U test_user -s "2021-09-10 00:00:00" -e "2021-09-10 23:00:00"
+###########################################################
+
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:16]: replace with two newlines.

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh [hunk 11]
```diff
@@ -0,0 +1,197 @@
+#!/bin/bash
+
+#############################################################
+
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:4]: remove blank line

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+read -s -p "Password to connect to Redshift cluster: " PGPASSWORD
+export  PGPASSWORD=$PGPASSWORD
+echo
+if [[ -z $PGCONNECT_TIMEOUT ]]; then
+  export PGCONNECT_TIMEOUT=10
+fi
+
+
+#TO: Validate ALL connection  inputs
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:41]: Implement or remove.

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+      d) export PGDATABASE=${OPTARG};;
+      U) export PGUSER=${OPTARG};;
+      s) STARTTIME=${OPTARG};;
+      e) ENDTIME=${OPTARG};;
+      t) export PGCONNECT_TIMEOUT=${OPTARG};;
+      esac
+done
+
+read -s -p "Password to connect to Redshift cluster: " PGPASSWORD
+export  PGPASSWORD=$PGPASSWORD
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:34]: remove extra space

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+  export PGCONNECT_TIMEOUT=10
+fi
+
+
+#TO: Validate ALL connection  inputs
+if [[ -z $PGHOST ]] || [[ -z $PGPORT ]] || [[ -z $PGDATABASE ]] || [[ -z $PGUSER ]] ; then
+echo "connection parameters required and cannot be empty, please enter correct  hostname, dbname, port-number, username to connect"      
+ exit -1
+fi
+
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:46]: reduce to 1 newline

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+
+if [ -s date.txt  ]; then
+STARTTIME = head -n 1  date.txt
+fi
+if [ -s date.txt  ]; then
+ENDTIME  = tail  -n 1  date.txt
+fi
+
+#echo $RUNTIMEQUERY
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:58]: remove if not needed

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+#echo $BOOST_MIN_RUNTIME
+#echo $BOOST_MAX_RUNTIME
+RESULT_RUNTIME=$(psql -c "$RUNTIMEQUERY" -A --tuples-only)
+if [ $CONN_STATUS -eq 2 ]; then
+echo "Connection failed. Please try again"
+exit $CONN_STATUS
+fi
+X=$(echo "$RESULT_RUNTIME" | awk -F'|' '{print $1F}')
+Y=$(echo "$RESULT_RUNTIME" | awk -F'|' '{print $NF}')
+#echo $X
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:193]: remove if not needed

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+if [[  -z $RESULT ]]; then
+echo "Please run the script with date interval on the test cluster where  you have executed queries with AQUA on and off"
+else 
+echo 'Query ID Redshift with AQUA turned on,Query ID Redshift with AQUA turned off,Runtime in seconds - AQUA turned on,Runtime in seconds - AQUA turned off,Speedup(colum C/Column B)' > output_file.csv
+echo $RESULT | tr ' ' '\n' | tr '|' ',' >> output_file.csv
+BOOST=$(echo "$RESULT" | awk -F'|' -v RS='\n' '{print $NF}' | sort -n);
+#echo $BOOST
+BOOST_MIN_RUNTIME=$(echo "$BOOST" | sed -n '1p')
+BOOST_MAX_RUNTIME=$(echo "$BOOST" | sed -n '$p')
+#echo $BOOST_MIN_RUNTIME
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:184]: remove if not needed.

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh [hunk 18]
```diff
@@ -0,0 +1,197 @@
...
+RESULT_RUNTIME=$(psql -c "$RUNTIMEQUERY" -A --tuples-only)
+if [ $CONN_STATUS -eq 2 ]; then
+echo "Connection failed. Please try again"
+exit $CONN_STATUS
+fi
+X=$(echo "$RESULT_RUNTIME" | awk -F'|' '{print $1F}')
+Y=$(echo "$RESULT_RUNTIME" | awk -F'|' '{print $NF}')
+#echo $X
+#echo $Y
+echo " The set of queries executed took $X in RedShift and took $Y in AQUA+RS.  The  performance gain AQUA for Redshift provided ranges from $BOOST_MIN_RUNTIME  to $BOOST_MAX_RUNTIME"
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:195]: extra spaces

**Review by sgromoll — CHANGES_REQUESTED** [review]
Please cleanup formatting a bit.

File: src/BlogContent/AQUA-Evaluation/aqua_capture_query.sh (hunk 1 above)
Comment (happycontribute) [reply src/BlogContent/AQUA-Evaluation/aqua_capture_query.sh:17]: done

File: src/BlogContent/AQUA-Evaluation/aqua_capture_query.sh (hunk 2 above)
Comment (happycontribute) [reply src/BlogContent/AQUA-Evaluation/aqua_capture_query.sh:49]: cleaned it

File: src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh (hunk 3 above)
Comment (happycontribute) [reply src/BlogContent/AQUA-Evaluation/aqua_execute_query.sh:15]: removed

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh (hunk 15 above)
Comment (happycontribute) [reply src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:58]: removed

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh (hunk 16 above)
Comment (happycontribute) [reply src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:193]: removed

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh (hunk 18 above)
Comment (happycontribute) [reply src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:195]: cleared spaces

File: src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh (hunk 14 above)
Comment (happycontribute) [reply src/BlogContent/AQUA-Evaluation/aqua_perf_compare.sh:46]: removed spaces

**Comment by happycontribute** [comment]
Please take a look at the readme file added and IP address are mocked in sample runs.

File: src/BlogContent/AQUA-Evaluation/Aqua_eligible_testqueries.sql [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+
+select count(*) from amazon_reviews WHERE product_title SIMILAR TO '%lap%';
+
+ select count(*) from amazon_reviews WHERE product_title SIMILAR TO '%lap%' group by star_rating ORDER BY star_rating desc;
+
+select count(customer_id), count(review_ID) from amazon_reviews where  review_body like '%one%car%';
+
+select count(*) from amazon_reviews where product_title ilike '%lap%' or product_title ilike '%e%|%E%' or customer_ID like '3%__%45__3';
+
+ SELECT customer_id , product_title , review_id  FROM amazon_reviews  WHERE product_title  SIMILAR TO ‘L%’ ORDER BY customer_id  LIMIT 20;
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/Aqua_eligible_testqueries.sql:15]: any reason there is inconsistent leading spaces and capitalization?   can we standard this throughout the file?  looks messy.

File: src/BlogContent/AQUA-Evaluation/Aqua_eligible_testqueries.sql [hunk 20]
```diff
@@ -0,0 +1,35 @@
+ select  count(*) from amazon_reviews where product_title SIMILAR TO  '%lap%' or
+product_title SIMILAR TO  '%iph%' or
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/Aqua_eligible_testqueries.sql:2]: two spaces here... this all looks haphazard, can you clean up formatting please.

File: src/BlogContent/AQUA-Evaluation/aqua_eligible_queries.sql [hunk 21]
```diff
@@ -0,0 +1 @@
...or product_title SIMILAR TO '%nice%' or product_title SIMILAR TO '%hope%';; select count(*) from amazon_reviews where product_title SIMILAR TO '%lap%' or product_title SIMILAR TO '%iph%' or product_title SIMILAR TO '%soa%' or product_title SIMILAR TO '%nice%' or product_title SIMILAR TO '%hope%';; select count(*) from amazon_reviews where product_title SIMILAR TO '%lap%' or product_title SIMILAR TO '%iph%';; select count(*) from amazon_reviews where product_title SIMILAR TO '%lap%' or product_title SIMILAR TO '%iph%';; select count(*) from amazon_reviews WHERE product_title SIMILAR TO '%lap%';; select count(*) from amazon_reviews WHERE product_title SIMILAR TO '%lap%';; select count(*) from amazon_reviews where product_title SIMILAR TO '%lap%' or product_title SIMILAR TO '%iph%' or product_title SIMILAR TO '%soa%' or product_title SIMILAR TO '%nice%' or product_title SIMILAR TO '%hope%';; select count(*) from amazon_reviews where product_title SIMILAR TO '%lap%' or product_title SIMILAR TO '%iph%';; select count(*) from amazon_reviews WHERE product_title SIMILAR TO '%lap%';; SELECT customer_id , product_title FROM amazon_reviews WHERE review_body SIMILAR TO '%(good|bad)%' limit 1;;
```
Comment (sgromoll) [inline src/BlogContent/AQUA-Evaluation/aqua_eligible_queries.sql:1]: double ;;?

**Review by sgromoll — CHANGES_REQUESTED** [review]

**Pull request merged**